tests/unit/testConfigs/locConstraintAssert.js

```diff
@@ -286,4 +286,18 @@ describe('locationConstraintAssert', () => {
         assert.strictEqual(locationConstraint.details.pathStyle, true,
             'pathstyle config should be true');
     });
+
+    it('should throw error if sizeLimitGB is not a number', () => {
+        const usEast1 = new LocationConstraint();
+        const locationConstraint = new LocationConstraint('aws_s3', true,
+            { sizeLimitGB: true });
+        assert.throws(() => {
+            locationConstraintAssert({
+                'us-east-1': usEast1,
+                'awsstoragesizelimit': locationConstraint,
+            });
+        },
+        '/bad config: locationConstraints[region].details.sizeLimitGB ' +
```

Comment:
I think a string does not work here (see https://nodejs.org/api/assert.html#assert_assert_throws_block_error_message)

Do you want a regex instead, starting with `/bad config...`? In which case should work but cannot be split across lines.

Reply:
The test passes :) It follows the same format as the rest of the tests in the file

Reply:
Here's what I mean:

```
jonathan@jonathan-XPS-L421X:~/git/ironman/backbeat$ node
> const assert = require('assert')
undefined
> > assert.throws(() => { throw new Error('foo'); }, '/bar')
... undefined
... > assert.throws(() => { throw new Error('foo'); }, /bar/)
... Error: foo
...     at assert.throws (repl:1:29)
...     at tryBlock (assert.js:305:5)
...     at innerThrows (assert.js:322:18)
...     at Function.throws (assert.js:350:3)
...     at repl:1:8
...     at sigintHandlersWrap (vm.js:22:35)
...     at sigintHandlersWrap (vm.js:73:12)
...     at ContextifyScript.Script.runInThisContext (vm.js:21:12)
...     at REPLServer.defaultEval (repl.js:340:29)
...     at bound (domain.js:280:14)
```

Reply:
i.e. `assert.throws` does not enforce the second argument check if it's a string

Reply:
And the rest of the tests in the file may be broken as well ;)

Reply:
rip

Reply:
Actually I just realize that it was in the same file where I saw such errors and added a `//TODO` in a previous PR to evolve to use `joi` for conf parsing.